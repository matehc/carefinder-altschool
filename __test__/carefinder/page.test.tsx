import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom"; 
import CareFinder from "@/app/carefinder/page";


describe("CareFinder component", () => {
    it("renders without crashing", () => {
      render(<CareFinder />);
    });
  
    it("displays the welcome heading", () => {
      render(<CareFinder />);
      const heading = screen.getByRole("heading", { level: 1 });
      expect(heading).toHaveTextContent("Welcome to CareFinder App");
    });
  
    it("renders the AltSchoolImage component", () => {
      render(<CareFinder />);
      const altSchoolImage = screen.getByTestId("altschool-image");
      expect(altSchoolImage).toBeInTheDocument();
    });
  
    it("renders the PlaceLocation component", () => {
      render(<CareFinder />);
      const placeLocation = screen.getByTestId("place-location");
      expect(placeLocation).toBeInTheDocument();
    });
  
    // Add more tests as needed
  
    it("matches the snapshot", () => {
      const { container } = render(<CareFinder />);
      expect(container).toMatchSnapshot();
    });
  });