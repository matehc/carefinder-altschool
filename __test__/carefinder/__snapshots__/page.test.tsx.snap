// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CareFinder component matches the snapshot 1`] = `
<div>
  <div
    class="text-[#1F4D36]  "
  >
    <div
      class="bg-[#1F4D36] "
    >
      <div
        class="text-gray-200 bg-[#1F4D36] py-3 text-center flex items-center justify-between max-w-[1500px] mx-auto"
      >
        <a
          class="cursor-pointer pl-2 md:pl-5 lg:pl-10"
          href="/"
        >
          <svg
            class=""
            fill=""
            height="83"
            viewBox="0 0 1080 1080"
            width="76"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M240.778 286.29H167.698L102.738 489H141.888L159.868 433.32H249.768L268.618 489H307.768L240.778 286.29ZM170.308 401.13L204.238 296.44L239.038 401.13H170.308ZM361.346 489.29V286.29H325.386V489.29H361.346ZM440.617 339.65V286.29H404.657V339.65H382.907V367.78H386.387C386.387 367.78 404.657 367.78 404.657 387.79V452.46C404.657 473.05 421.187 489.29 441.487 489.29H466.137V457.39H451.927C445.837 457.39 440.617 452.17 440.617 446.08V390.4C440.617 368.94 395.087 367.78 387.547 367.78H466.137V339.65H440.617ZM216.998 647.26C216.998 647.26 178.138 638.27 176.978 637.98C163.928 634.5 152.908 628.12 152.908 614.78C152.908 600.28 170.598 588.39 192.928 588.39C220.188 588.39 235.558 605.21 235.558 621.74H273.548C273.548 586.36 245.998 556.49 192.928 556.49C147.978 556.49 114.918 580.56 114.918 615.07C114.918 650.74 137.538 667.85 170.598 675.39C170.598 675.39 209.748 684.38 210.618 684.67C225.698 688.44 238.168 695.98 238.168 710.77C238.168 729.04 215.548 740.64 191.768 738.61C159.578 736 147.688 714.83 147.688 699.75H109.988C109.988 732.81 137.538 769.35 192.348 770.8C233.528 771.67 276.158 749.05 276.158 710.77C276.158 675.39 251.798 655.67 216.998 647.26ZM368.566 736.87C345.656 736.87 333.476 715.12 333.476 690.18C333.476 664.95 348.556 643.49 368.276 643.49C383.936 643.49 396.406 655.09 400.756 671.04H439.616C433.236 633.63 403.366 611.59 367.986 611.59C328.256 611.59 293.746 646.39 295.776 694.82C297.516 740.06 329.996 768.48 368.856 768.48C405.106 768.48 432.076 744.41 439.036 711.64H399.016C395.826 724.69 384.226 736.87 368.566 736.87ZM541.295 612.17C519.255 613.91 500.405 636.82 500.115 677.13C500.405 657.12 509.395 642.62 529.985 642.62C552.605 642.62 559.855 659.44 559.855 678V765.29H596.105V665.82C596.105 631.02 567.395 610.14 541.295 612.17ZM464.155 562.29V765.29H500.115V562.29H464.155ZM693.71 611.59C655.14 611.59 619.76 640.3 619.76 690.18C619.76 740.06 654.56 768.19 693.71 768.48C733.15 768.19 767.95 739.77 767.95 690.18C767.95 640.3 732.28 611.59 693.71 611.59ZM693.71 736.58C674.57 736.58 656.01 721.5 656.01 690.18C656.01 660.6 674.28 643.78 693.71 643.49C713.43 643.78 731.7 660.89 731.7 690.18C731.7 721.5 713.14 736.58 693.71 736.58ZM860.8 611.59C822.23 611.59 786.85 640.3 786.85 690.18C786.85 740.06 821.65 768.19 860.8 768.48C900.24 768.19 935.04 739.77 935.04 690.18C935.04 640.3 899.37 611.59 860.8 611.59ZM860.8 736.58C841.66 736.58 823.1 721.5 823.1 690.18C823.1 660.6 841.37 643.78 860.8 643.49C880.52 643.78 898.79 660.89 898.79 690.18C898.79 721.5 880.23 736.58 860.8 736.58ZM996.57 765.29V562.29H960.61V765.29H996.57ZM996.998 445H504.998V489H996.998V445Z"
              fill="white"
            />
          </svg>
        </a>
        <h1
          class="text-[14px] hidden md:text-[32px] md:block"
        >
          Welcome to CareFinder App
        </h1>
        <div
          class="mr-4 rounded-full h-[60px] w-[60px] md:mr-5 md:h-[40px] md:w-[40px] lg:h-[60px] lg:w-[60px] lg:mr-10"
        >
          <img
            alt="Picture of the author"
            class="rounded-full h-[60px] w-[60px] md:h-[40px] md:w-[40px] lg:h-[60px] lg:w-[60px]"
            data-nimg="1"
            decoding="async"
            height="83"
            loading="lazy"
            src="/_next/image?url=https%3A%2F%2Fencrypted-tbn0.gstatic.com%2Fimages%3Fq%3Dtbn%3AANd9GcSGGT_X83CRh2jFX9iojbv3ByMRcm4TDDAnIA%26usqp%3DCAU&w=256&q=75"
            srcset="/_next/image?url=https%3A%2F%2Fencrypted-tbn0.gstatic.com%2Fimages%3Fq%3Dtbn%3AANd9GcSGGT_X83CRh2jFX9iojbv3ByMRcm4TDDAnIA%26usqp%3DCAU&w=96&q=75 1x, /_next/image?url=https%3A%2F%2Fencrypted-tbn0.gstatic.com%2Fimages%3Fq%3Dtbn%3AANd9GcSGGT_X83CRh2jFX9iojbv3ByMRcm4TDDAnIA%26usqp%3DCAU&w=256&q=75 2x"
            style="color: transparent;"
            width="76"
          />
        </div>
      </div>
    </div>
    <div
      class=" max-w-[1400px] mx-auto"
    >
      <div
        class="px-4"
      >
        <div
          class="mt-8 text-[10px] flex justify-between items-center md:text-[20px]"
        >
          <p
            class="mt-0  text-[16px] md:text-[20px] md:mt-8"
          >
            Select your Location
          </p>
          <div>
            <a
              class="downloadbtn p-3 bg-[#1F4D36] text-white rounded-md"
              download="hospital_location.csv"
              href="data:text/csv;charset=utf-8,ï»¿"
              target="_self"
            >
              Export to CSV
            </a>
          </div>
        </div>
        <select
          class="w-full p-4 border-[.23px] border-[#1F4D36] my-4 "
          id="location"
          name="location"
        >
          <option>
            Search by Location
          </option>
          <option>
            lagos
          </option>
          <option>
            ibadan
          </option>
          <option>
            abuja
          </option>
          <option>
            kano
          </option>
          <option>
            jos
          </option>
          <option>
            kaduna
          </option>
          <option>
            enugu
          </option>
          <option>
            owerri
          </option>
          <option>
            portharcourt
          </option>
          <option>
            calabar
          </option>
          <option>
            asaba
          </option>
          <option>
            warri
          </option>
          <option>
            benin
          </option>
          <option>
            akure
          </option>
          <option>
            ilorin
          </option>
          <option>
            ibadan
          </option>
          <option>
            abakaliki
          </option>
          <option>
            uyo
          </option>
          <option>
            makurdi
          </option>
          <option>
            minna
          </option>
          <option>
            zaria
          </option>
          <option>
            bauchi
          </option>
          <option>
            yola
          </option>
          <option>
            sokoto
          </option>
          <option>
            katsina
          </option>
          <option>
            gusau
          </option>
          <option>
            jalingo
          </option>
          <option>
            lafia
          </option>
          <option>
            damaturu
          </option>
          <option>
            maiduguri
          </option>
          <option>
            awka
          </option>
          <option>
            onitsha
          </option>
          <option>
            umahia
          </option>
          <option>
            bayelsa
          </option>
          <option>
            yenagoa
          </option>
          <option>
            abakaliki
          </option>
          <option>
            nsukka
          </option>
          <option>
            nsit ibom
          </option>
          <option>
            ikot ekpene
          </option>
          <option>
            ikom
          </option>
          <option>
            orlu
          </option>
          <option>
            ogbomosho
          </option>
          <option>
            abeokuta
          </option>
          <option>
            sagamu
          </option>
          <option>
            ijebu ode
          </option>
          <option>
            oshogbo
          </option>
          <option>
            ile ife
          </option>
          <option>
            ilesha
          </option>
          <option>
            akure
          </option>
          <option>
            ondo
          </option>
        </select>
        <div
          class="flex gap-4 my-8"
        >
          <div
            class="w-full md:w-1/2"
          >
            <div>
              <form
                class="flex items-center"
              >
                <label
                  class="sr-only"
                  for="simple-search"
                >
                  Search
                </label>
                <div
                  class="relative w-full undefined"
                >
                  <div
                    class="absolute inset-y-0 left-0 flex items-center pl-3 pointer-events-none"
                  />
                  <input
                    class="w-full p-2.5 pl-5 text-gray-500 bg-white border rounded-md shadow-sm outline-none appearance-none focus:border-indigo-600 undefined"
                    placeholder="Filter by Hospital Name"
                    type="text"
                  />
                </div>
              </form>
            </div>
          </div>
          <div
            class="w-full md:w-1/2"
          >
            <div>
              <form
                class="flex items-center"
              >
                <label
                  class="sr-only"
                  for="simple-search"
                >
                  Search
                </label>
                <div
                  class="relative w-full undefined"
                >
                  <div
                    class="absolute inset-y-0 left-0 flex items-center pl-3 pointer-events-none"
                  />
                  <input
                    class="w-full p-2.5 pl-5 text-gray-500 bg-white border rounded-md shadow-sm outline-none appearance-none focus:border-indigo-600 undefined"
                    placeholder="Filter by Email"
                    type="text"
                  />
                </div>
              </form>
            </div>
          </div>
          <div
            class="w-full md:w-1/2"
          >
            <div>
              <form
                class="flex items-center"
              >
                <label
                  class="sr-only"
                  for="simple-search"
                >
                  Search
                </label>
                <div
                  class="relative w-full undefined"
                >
                  <div
                    class="absolute inset-y-0 left-0 flex items-center pl-3 pointer-events-none"
                  />
                  <input
                    class="w-full p-2.5 pl-5 text-gray-500 bg-white border rounded-md shadow-sm outline-none appearance-none focus:border-indigo-600 undefined"
                    placeholder="Filter by Latitude/Longitude"
                    type="text"
                  />
                </div>
              </form>
            </div>
          </div>
        </div>
        <p
          class="mt-8 text-[16px] md:text-[20px]"
        >
          List of Hospitals around 
        </p>
        <div
          class=" "
        >
          <div
            class="w-full overflow-x-auto block rounded-[0.25rem] min-h-[30.3rem] bg-white  shadow-shadowOne border border-[#213f7d0f]"
          >
            <table
              class="w-full h-full rounded-[0.25rem]   font-medium text-small"
            >
              <thead
                class="text-xs"
              >
                <tr>
                  <th
                    class="py-4 px-5 text-left font-bold uppercase text-sm"
                  >
                    S/N
                  </th>
                  <th
                    class="py-4 px-5 text-left font-bold uppercase text-sm"
                  >
                    <span>
                      HOSPITAL NAME
                    </span>
                  </th>
                  <th
                    class="py-4 px-5 text-left font-bold uppercase text-sm"
                  >
                    <span>
                      EMAIL
                    </span>
                  </th>
                  <th
                    class="py-4 px-5 text-left font-bold uppercase text-sm"
                  >
                    <span>
                      ADDRESS
                    </span>
                  </th>
                  <th
                    class="py-4 px-5 text-left font-bold uppercase text-sm"
                  >
                    <span>
                      REGION
                    </span>
                  </th>
                  <th
                    class="py-4 px-5 text-left font-bold uppercase text-sm"
                  >
                    <span>
                      LONG/LAT
                    </span>
                  </th>
                  <th
                    class="py-4 px-5 text-left font-bold uppercase text-sm"
                  >
                    <span>
                      TIME ZONE
                    </span>
                  </th>
                </tr>
              </thead>
              <tbody />
            </table>
          </div>
          <div
            class="flex items-center gap-x-8 justify-between px-4 py-5  rounded-b-[0.5rem]"
          >
            <span
              class="flex items-center gap-1 text-[#060F21] text-small text-epps-stalegray-500"
            >
              <div
                class="hidden sm:block"
              >
                Showing
              </div>
              1
               of 
              0
            </span>
            <div
              class="space-x-5"
            >
              <button
                class="bg-[#213f7d1a] p-1 rounded-[.25rem] cursor-pointer"
                disabled=""
              >
                <svg
                  color="#213f7d66"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  style="color: rgba(33, 63, 125, 0.4);"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <polyline
                    fill="none"
                    points="9 6 15 12 9 18"
                    stroke="#000"
                    stroke-width="2"
                    transform="matrix(-1 0 0 1 24 0)"
                  />
                </svg>
              </button>
              <span
                class="text-epps-stalegray-500"
              >
                1
              </span>
              <button
                class="bg-[#213f7d1a] p-1 rounded-[.25rem] cursor-pointer"
                disabled=""
              >
                <svg
                  color="#213f7d66"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  style="color: rgba(33, 63, 125, 0.4);"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0V0z"
                    fill="none"
                  />
                  <path
                    d="M10.02 6L8.61 7.41 13.19 12l-4.58 4.59L10.02 18l6-6-6-6z"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
